pyobjc-core 9.0

skupr-anaconda opened the pull request:
**Jira ticket:** [PKG-816](https://anaconda.atlassian.net/browse/PKG-816) (pyobjc-core 9.0)

**The upstream data:**
The upstream:  https://github.com/ronaldoussoren/pyobjc/tree/v9.0/pyobjc-core
License: https://github.com/ronaldoussoren/pyobjc/blob/v9.0/pyobjc-core/License.txt
Requirements:
- https://github.com/ronaldoussoren/pyobjc/blob/v9.0/pyobjc-core/setup.cfg
- https://github.com/ronaldoussoren/pyobjc/blob/v9.0/pyobjc-core/setup.py

**_Actions:_**

1. Update a patch
2. Skip `py<37`
3. Add `libffi 3.4` pinning to `host` and update pinning in `run`
4. Add a `description` 

**Package's statistics**
<details>

 * Priority D | effort: easy | Category: other | subcategory: dependency | pkg_type: python
 * Outdated platfroms: 'osx-64', 'osx-arm64'
 * Latest version: 9.0
 * Release on PyPi:
    * version: 9.0
    * date: 2022-11-05T19:17:53
* [PyPi history](https://pypi.org/project/pyobjc-core/#history)
 * Popularity: 
    * 3 months downloads: 33973.0
    * All time:  109948 downloads -  pyobjc-core  9.0  Python<->ObjC Interoperability Module  

</details>

**Other checks:**

<details>

5. - [x] https://github.com/ronaldoussoren/pyobjc is present
6. - [x] Check the pinnings
7. - [x] `dev_url` is present
    https://github.com/ronaldoussoren/pyobjc/tree/master/pyobjc-core
8. - [x] `doc_url` is present
    https://github.com/ronaldoussoren/pyobjc/blob/master/pyobjc-core/README.txt
9. - [x] Verify that the `build_number` is correct
10. - [x] has `setuptools`
11. - [x] has `wheel`
12. - [x] `pip` in test
13. - [x] Verify the test section
14. - [x] Verify if the package is `architecture specific`
15. - [x] Verify that private modules are not mentioned in the recipe For example: (_private_module)
16.  - [x] license_file: License.txt is present
17. - [x] License: MIT
    - [x] License is `spdx` compliant
 * Check if the license identifier has correct name from the SPDX License List

| Identifier         |
|:-------------------|
| MIT                |
| MIT-0              |
| MIT-advertising    |
| MIT-CMU            |
| MIT-enna           |
| MIT-feh            |
| MIT-Modern-Variant |
| MIT-open-group     |
| MITNFA             |

18. - [x] license_family MIT is present
</details>



**Links:**
* [Anaconda Recipes feedstock](https://github.com/AnacondaRecipes/pyobjc-core-feedstock)
* [Update branch](https://github.com/AnacondaRecipes/pyobjc-core-feedstock/tree/9.0)
* [conda-forge recipe](https://github.com/{upstream}/{feedstock_name}-feedstock)
* [PyPI](https://pypi.org/project/pyobjc-core)
* [Diff between upstream and feature branch](https://github.com/conda-forge/pyobjc-core-feedstock/compare/main...AnacondaRecipes:9.0)
* [Diff between origin and feature branch](https://github.com/AnacondaRecipes/pyobjc-core-feedstock/compare/master...AnacondaRecipes:9.0)
* [The last merged PRs](https://github.com/AnacondaRecipes/pyobjc-core-feedstock/pulls?q=is%3Apr+is%3Amerged+sort%3Aupdated-desc+)

**Updating the recipe:**
If the recipe needs additional modification the update branch can be modified. Note that the PR diffs are not updated with these changes.
```
git clone -b 9.0 git@github.com:AnacondaRecipes/pyobjc-core-feedstock.git
```

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_edw21iq_py/clone/recipe/meta.yaml:57: missing_license_url: The recipe is missing a license_url
WARNING: /tmp/abs_edw21iq_py/clone/recipe/meta.yaml:57: missing_doc_source_url: The recipe is missing a doc_source_url
WARNING: /tmp/abs_edw21iq_py/clone/recipe/meta.yaml:57: missing_description: The recipe is missing a description
Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_79df1mjsym/clone/recipe/meta.yaml:63: missing_doc_source_url: The recipe is missing a doc_source_url
WARNING: /tmp/abs_79df1mjsym/clone/recipe/meta.yaml:63: missing_license_url: The recipe is missing a license_url
Warnings were found

</details>

cbouss reviewed (approved):
LGTM

varlackc reviewed (approved):
LGTM

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_7cueyv6r1z/clone/recipe/meta.yaml:63: missing_doc_source_url: The recipe is missing a doc_source_url
WARNING: /tmp/abs_7cueyv6r1z/clone/recipe/meta.yaml:63: missing_license_url: The recipe is missing a license_url
Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_06i3va8ki9/clone/recipe/meta.yaml:63: missing_license_url: The recipe is missing a license_url
WARNING: /tmp/abs_06i3va8ki9/clone/recipe/meta.yaml:63: missing_doc_source_url: The recipe is missing a doc_source_url
Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_cb_9lpbf60/clone/recipe/meta.yaml:63: missing_license_url: The recipe is missing a license_url
WARNING: /tmp/abs_cb_9lpbf60/clone/recipe/meta.yaml:63: missing_doc_source_url: The recipe is missing a doc_source_url
Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_31yvizbsip/clone/recipe/meta.yaml:65: missing_doc_source_url: The recipe is missing a doc_source_url
WARNING: /tmp/abs_31yvizbsip/clone/recipe/meta.yaml:65: missing_license_url: The recipe is missing a license_url
Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_degl3xg3kz/clone/recipe/meta.yaml:65: missing_doc_source_url: The recipe is missing a doc_source_url
WARNING: /tmp/abs_degl3xg3kz/clone/recipe/meta.yaml:65: missing_license_url: The recipe is missing a license_url
Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_a7fm8vp10o/clone/recipe/meta.yaml:65: missing_license_url: The recipe is missing a license_url
WARNING: /tmp/abs_a7fm8vp10o/clone/recipe/meta.yaml:65: missing_doc_source_url: The recipe is missing a doc_source_url
Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_43swbyznsr/clone/recipe/meta.yaml:65: missing_license_url: The recipe is missing a license_url
WARNING: /tmp/abs_43swbyznsr/clone/recipe/meta.yaml:65: missing_doc_source_url: The recipe is missing a doc_source_url
Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_5cy57g68b1/clone/recipe/meta.yaml:65: missing_license_url: The recipe is missing a license_url
WARNING: /tmp/abs_5cy57g68b1/clone/recipe/meta.yaml:65: missing_doc_source_url: The recipe is missing a doc_source_url
Warnings were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_80epq1b57b/clone/recipe/meta.yaml:65: missing_doc_source_url: The recipe is missing a doc_source_url
WARNING: /tmp/abs_80epq1b57b/clone/recipe/meta.yaml:65: missing_license_url: The recipe is missing a license_url
Warnings were found

</details>

skupr-anaconda commented:
I've updated `MACOS_SDK_VERSION` and added `CONDA_BUILD_SYSROOT` for `osx64`:
```
MACOSX_SDK_VERSION:       # [osx]
  - "11.3"                  # [osx and arm64]
  # SDK >=10.14 is needed on osx64 to prevent using the flag -Wno-unused-parameter, 
  # see https://github.com/ronaldoussoren/pyobjc/blob/595adfb6a08b41da8d565bb28aa508b22c946e46/pyobjc-core/setup.py#L607
  - "10.14"                  # [osx and x86_64]
CONDA_BUILD_SYSROOT:       # [osx and x86_64]
  - /opt/MacOSX10.14.sdk   # [osx and x86_64]
```

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

WARNING: /tmp/abs_aa2ued5uud/clone/recipe/meta.yaml:65: missing_license_url: The recipe is missing a license_url
WARNING: /tmp/abs_aa2ued5uud/clone/recipe/meta.yaml:65: missing_doc_source_url: The recipe is missing a doc_source_url
Warnings were found

</details>
